tests/generate_sources_test.py

```diff
@@ -0,0 +1,592 @@
+
+        # Get connector
+        conn = get_connector_redshift(
+            host=os.environ["HOST_REDSHIFT"],
+            user=os.environ["USER_REDSHIFT"],
+            password=os.environ["PASSWORD_REDSHIFT"],
+            database=database,
+        )
+
+        with conn.cursor() as cursor:
+            cursor.execute(f"CREATE SCHEMA IF NOT EXISTS {schema};")
+            with open(input["create_model_sql_file"], "r") as sql_file:
+                query = sql_file.read()
+            cursor.execute(query)
+            with open(input["insert_data_sql_file"], "r") as sql_file:
+                query = sql_file.read()
+            cursor.execute(query)
+            conn.commit()
+            cursor.execute(f"SELECT * FROM {schema}.{table} LIMIT 1;")
+            result = cursor.fetchall()
+        assert len(result) == 1
+        conn.close()
+
+    elif input["adapter"] == "bigquery":
+        # Check env vars
+        assert "PROJECT_BIGQUERY" in os.environ
+        assert "SERVICE_ACCOUNT_GCP" in os.environ
+
+        # Get env vars
+        project_id = os.environ["PROJECT_BIGQUERY"]
+        sa_key = os.environ["SERVICE_ACCOUNT_GCP"]
+
+        database = input["settings"]["database"]
+        schema = input["settings"]["schema"]
+        table = input["settings"]["table"]
+
+        # Get client
+        client = get_client_bigquery(sa_key, project_id)
+
+        # Generate data
+        query_job = client.query(
+            f"CREATE SCHEMA IF NOT EXISTS `{project_id}.{schema}`;"
+        )
+        query_job.result()
+        assert query_job.errors == None
+        with open(input["create_model_sql_file"], "r") as sql_file:
+            query = sql_file.read()
+        query_job = client.query(query, location="US")
+        query_job.result()
+        assert query_job.errors == None
+        with open(input["insert_data_sql_file"], "r") as sql_file:
+            query = sql_file.read()
+        query_job = client.query(query, location="US")
+        query_job.result()
+        assert query_job.errors == None
+        query_job = client.query(f"SELECT * FROM {schema}.{table} LIMIT 1;")
+        query_job.result()
+        rows = []
+        assert query_job.errors == None
+        for row in query_job:
+            rows.append(row)
+        assert len(rows) == 1
+        client.close()
+    else:
+        raise Exception("Adapter not supported")
+
+
+# Generate sources cases
+
+cases_list_generate_sources = []
+
+for case in cases_list:
+    id = case["id"]
+    cases_list_generate_sources.append(
+        pytest.param(
+            case["input"],
+            id=case["id"],
+            marks=pytest.mark.dependency(
+                name=f"test_generate_sources[{id}]",
+                depends=[f"test_generate_data[{id}]"],
+            ),
+        )
+    )
+
+
+@pytest.mark.parametrize("input", cases_list_generate_sources)
+def test_generate_sources(input):
+    # Generate sources command
+    command = [
+        "dbt-coves",
```

Comment:
Isn't the argument `--profiles-dir` missing here to point to the test_case's folder?